### PR TITLE
apt: add update_cache_allow_releaseinfo_change option

### LINKED ISSUE #30754: Failed to update apt cache (update_cache)
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

apt

##### ANSIBLE VERSION

```
ansible 2.2.0.0
```

##### OS / ENVIRONMENT

Ubuntu 14.04

##### SUMMARY

Cannot update cache.

##### STEPS TO REPRODUCE

```yaml
---
- hosts: localhost
  gather_facts: no
  connection: local
  become: yes
  tasks:
    - apt: update_cache=yes
```

##### ACTUAL RESULTS

```
PLAY [localhost] ***************************************************************

TASK [apt] *********************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to update apt cache."}	

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1
```

### PR DESCRIPTION
**Pull request by agix**
##### SUMMARY
This add an option to pass `--allow-releaseinfo-change` to `update_cache`

It should fixes https://github.com/ansible/ansible/issues/30754

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
Change in apt module

### CODE REVIEW

File: lib/ansible/modules/packaging/os/apt.py
Comment (bcoca) [inline lib/ansible/modules/packaging/os/apt.py]: remove update_cache from name and add to the description 'this option is only used if update_cache is yes/truye'

**Review by bcoca — CHANGES_REQUESTED** [review]

**Comment by mkosobucki** [comment]
any update on this?

**Comment by asper-ru** [comment]
No any movement yet, eh?
Do we wait for review from @bcoca ?

**Comment by bcoca** [comment]
needs a change log ... i would also like to add tests but not sure how to do that without possibly breaking the apt based images

**Comment by freeyoung** [comment]
Since Debian `bullseye` was released the day before yesterday, I guess there would be more Ansible users wishing this to be merged earlier.

**Review by mds-lord — APPROVED** [review]
this option is truly missing and should be added to apt module.

**Comment by tagirb** [comment]
Came across this issue just today. Any news as to when this PR will be merged?

**Comment by nitzmahone** [comment]
Closing for inactivity- feel free to reopen if anyone wants to run this one to completion

**Pull request closed without merging**
